plutus-ledger-api/src/Plutus/V1/Ledger/Tx.hs

```diff
@@ -207,7 +210,7 @@ isPubKeyOut = isJust . txOutPubKey
 
 -- | Whether the output is a pay-to-script output.
 isPayToScriptOut :: TxOut -> Bool
-isPayToScriptOut = isJust . txOutDatum
+isPayToScriptOut = isJust . txOutValidatorHash
```

Comment:
Isn't this change unrelated to the CIP? It looks more like a fix to make the type of the address (p2sh or normal) to be determined not by the existence of the datum, but by the "deserialization" / format of the address

Reply:
Yeah, its a bug fix. The previous version was wrong.